Source/santactl/Commands/SNTCommandDoctor.m

```diff
@@ -0,0 +1,133 @@
+/// Copyright 2025 North Pole Security, Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     https://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+
+#import <Foundation/Foundation.h>
+#include <unistd.h>
+
+#include <libproc.h>
+
+#import "Source/common/SNTConfigurator.h"
+#import "Source/common/SNTLogging.h"
+#import "Source/santactl/SNTCommand.h"
+#import "Source/santactl/SNTCommandController.h"
+
+@interface SNTCommandDoctor : SNTCommand <SNTCommandProtocol>
+@end
+
+void print(NSString *format, ...) {
+  va_list args;
+  va_start(args, format);
+
+  if (isatty(STDOUT_FILENO)) {
+    if ([format hasPrefix:@"[-]"]) {
+      format = [NSString stringWithFormat:@"\033[31m%@\033[0m\n", format];
+    } else if ([format hasPrefix:@"[+]"]) {
+      format = [NSString stringWithFormat:@"\033[32m%@\033[0m\n", format];
+    } else {
+      format = [NSString stringWithFormat:@"%@\n", format];
+    }
+  }
+  vfprintf(stdout, format.UTF8String, args);
+  va_end(args);
+}
+
+@implementation SNTCommandDoctor
+
+REGISTER_COMMAND_NAME(@"doctor")
+
++ (BOOL)requiresRoot {
+  // This command requires root in order to retrieve all the data it needs.
+  return YES;
+}
+
++ (BOOL)requiresDaemonConn {
+  return NO;
+}
+
++ (NSString *)shortHelpText {
+  return @"TODO";
+}
+
++ (NSString *)longHelpText {
+  return @"TODO";
+}
+
+- (void)runWithArguments:(NSArray *)arguments {
+  BOOL err = NO;
+  err |= [self validateProcesses];
+  err |= [self validateConfiguration];
+  exit(err);
+}
+
+- (BOOL)validateProcesses {
+  print(@"=> Validating processes...");
+
+  BOOL foundSanta = NO, foundSantad = NO, foundSantaSyncService = NO;
+
+  // Get the number of procs and then allocate a buffer large enough to receive
+  // data about all of them.
+  int n = proc_listpids(PROC_ALL_PIDS, 0, NULL, 0);
+  int *buffer = (int *)malloc(sizeof(int) * n);
```

Comment:
The number returned by `proc_listpids` is already multiplied by `sizeof(int)`:  https://github.com/apple-oss-distributions/xnu/blob/8d741a5de7ff4191bf97d57b9f54c2f6d4a15585/bsd/kern/proc_info.c#L375-L379

(looks like you account for this after the second call in the for loop conditional).

As a tangent, likely out of scope for this change, but we do this `proc_listpids` dance already in the proc tree. Would be kinda nice to have some shared interface https://github.com/google/santa/blob/main/Source/santad/ProcessTree/process_tree_macos.mm#L140

Reply:
nit: I know this is a short live process that is about to exit, but would be nice to free the buffer,